milli/src/heed_codec/facet/field_id_codec.rs

```diff
@@ -0,0 +1,28 @@
+use std::borrow::Cow;
+use std::convert::TryInto;
+
+use heed::zerocopy::AsBytes;
+
+use crate::{FieldId, BEU16};
+
+pub struct FieldIdCodec;
+
+impl<'a> heed::BytesDecode<'a> for FieldIdCodec {
+    type DItem = FieldId;
+
+    fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem> {
+        let bytes: [u8; 2] = bytes[..2].try_into().ok()?;
+        let field_id = BEU16::from(bytes).get();
+        Some(field_id)
+    }
+}
+
+impl<'a> heed::BytesEncode<'a> for FieldIdCodec {
+    type EItem = FieldId;
+
+    fn bytes_encode(field_id: &Self::EItem) -> Option<Cow<[u8]>> {
+        let field_id = BEU16::new(*field_id);
+        let bytes = field_id.as_bytes();
+        Some(Cow::Owned(bytes.to_vec()))
+    }
+}
```

Comment:
Why not directly declare a `type FieldIdCodec = BEU16` here, instead?

Reply:
When I looked into it, I found that `BEU16` didn't implement `BytesEncode`/`Decode`.

Reply:
You should probably [use `OwnedType<BEU16>` instead sorry](https://docs.rs/heed-types/0.8.0/heed_types/struct.OwnedType.html).